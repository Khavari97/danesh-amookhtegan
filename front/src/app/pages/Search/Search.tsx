import React, { Component } from 'react';
import { Container, Typography } from "@material-ui/core";
import CustomTabs, { TabBarProps } from "../../components/customs/CustomTabs/CustomTabs";
import ProjectResult from "./ProjectResult";
import UserResult from "./UserResult";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


class Search extends Component<any> {

    SearchTabs: TabBarProps[] = [
        { label: "پروژه", component: <ProjectResult /> },
        { label: "افراد", component: <UserResult /> },
    ]

    render() {
        return (
            <Container>
                {this.props.homeVersion ? "" : <Typography data-testid="h1" variant={"h1"} component={"h1"}>جستجو</Typography>}
                <CustomTabs tabs={this.SearchTabs} />
            </Container>
        )
    }
}


export default Search;