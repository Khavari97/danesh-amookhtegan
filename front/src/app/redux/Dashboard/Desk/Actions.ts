import {DESK_LOADING, DESK_USER_PROJECTS, DESK_USER_REQUESTS, IProjectCard, IRequestedCard} from "./ActionTypes";

export const userProjectAction = (projects: IProjectCard[]) => {
    return {
        type: DESK_USER_PROJECTS,
        payload: projects,
        loading: false,
    }
}

export const userRequestAction = (requests: IRequestedCard[]) => {
    return {
        type: DESK_USER_REQUESTS,
        payload: requests,
        loading: false,
    }
}

export const deskLoading = () => {
    return {
        type: DESK_LOADING,
    }
}