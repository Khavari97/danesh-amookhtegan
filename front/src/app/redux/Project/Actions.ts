import {IProjectStatus, SET_PROJECT_SLUG} from "./ActionTypes";

export const setProjectAction = (slug: string, status: IProjectStatus) => {
    return {
        type: SET_PROJECT_SLUG,
        payload: {
            slug,
            status
        }
    }
}