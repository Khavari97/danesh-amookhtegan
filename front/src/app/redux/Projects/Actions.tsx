import {Dispatch} from "redux";
import {apiClient} from "../../api/_api";
import {Get_Projects} from "./ActionTypes";

export const GetProjects = () =>

    (dispatch: Dispatch) => {
        try {
            apiClient.get("dashboard/projects/", {consoleMessage: true})
                .then((response => {
                    dispatch({
                        type: Get_Projects,
                        projects: response.results
                    })

                })).catch((error => {

            }))

        } catch {

        }


    }


