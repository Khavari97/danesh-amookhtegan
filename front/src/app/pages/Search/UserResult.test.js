import React from 'react';
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../redux/store";
import UserResult from "./UserResult";
import renderer from "react-test-renderer";
import Search from "./Search";
import sinon from "sinon";
import { render, screen, prettyDOM, act, waitFor, wait, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store';
import { FilterUsers, searchUsersPage, setSearchText } from "../../redux/Search/Actions";
import thunk from 'redux-thunk'
import * as actions from "../../redux/Search/Actions"

const middlewares = [thunk]
const mockStore = configureStore(middlewares);


Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (props = {}, state = null) => {
  return shallow(
    <Provider store={store}>
      <UserResult {...props} />
    </Provider>

  );
};
const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};
describe(" test of ProjectResult page ", () => {
  test("render without error", () => {
    let wrapper = setup();
    expect(wrapper).toBeTruthy();
    expect(wrapper.length).toEqual(1);
  })

  test('show Users : ', () => {
    const wrapper = setup();
    let field = findByTestAttr(wrapper, 'showUsers')
    expect(field.length).toBe(0);
    expect(field.text).toBeTruthy();
  })

  it('check props not render without class', () => {
    let wrapper = setup();

    let props = {
      label: 'button', onClick: () => { }
    }

    wrapper.setProps(props)

    expect(wrapper.props()).not.toEqual(props);
    expect(wrapper.props().label).toBeUndefined()

  })

  it('simulate button', () => {
    let wrapper = setup();
    const mockCallBack = sinon.spy();
    let props = {
      label: 'button', onClick: mockCallBack
    }
    wrapper.setProps(props)
    wrapper.update();
    expect(mockCallBack).toHaveProperty('callCount', 0);
  })

  test('show counter: ', () => {
    let wrapper = setup();
    wrapper.setProps({
      location: 'string',
      page: 'number'
    })
    expect(wrapper.length).not.toEqual(0);
  })

  test('snapshot', () => {
    let wrapper =(<BrowserRouter>
    <Provider store={store}>
      <UserResult/>
    </Provider></BrowserRouter>)
    const snapshot = renderer.create(wrapper).toJSON();
    expect(snapshot).toMatchSnapshot();
})

})


export function createTestStore() {
  const store = createStore(
    combineReducers({
      Users: userReducer,
    })
  );
  return store;
}


describe('integration test component', () => {

  let store_mock

  beforeEach(() => {
    store_mock = mockStore({
      search: {
        users: [
          {
            avatar: "http://api.planb.markop.ir/media/defaults/default.png",
            city: "تهران",
            description: "دانشجوی دانشگاه خوارزمی ترم 6",
            firstName: "آتنا",
            lastName: "گنجی",
            province: "تهران",
            url: "/users/atena/"
          }
        ]
        , text: ""
      },
      userProfile: {
        loading: true,
        provinces: [],
        cities: [],
        universities: [],
        profile: {},
        description: '',
        data: {},
        parentSkills: [],
        skills: [],
        resume: null,
        avatar: null,
        briefprofile: {}
      }
    });
  });

  test("mock store check users", () => {
    const dom = render(
      <BrowserRouter>
        <Provider store={store_mock}>
          <UserResult />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getAllByTestId("card_user").length).toEqual(1)
  })


  test("change search input", async () => {
    const spySetSearchText = jest.spyOn(actions, 'setSearchText');
    store_mock.dispatch = jest.fn()
    const dom = render(
      <BrowserRouter>
        <Provider store={store_mock}>
          <UserResult />
        </Provider>
      </BrowserRouter>
    );

    fireEvent.change(screen.getByTestId('search-input').querySelector('input'), { target: { value: 'test' } });
    expect(screen.getByTestId('search-input').querySelector('input').value).toBe('test');

    //check dispatch one setSearchText after input change
    await wait(() => {
      expect(spySetSearchText).toHaveBeenCalledTimes(1);
    });
  })

  test("change pagination ", async () => {
    const spyChangePagination = UserResult.prototype.handleChange = jest.fn();
    const spySearchUserPage = jest.spyOn(actions, 'searchUsersPage');
    store_mock.dispatch = jest.fn()
    const dom = render(
      <BrowserRouter>
        <Provider store={store_mock}>
          <UserResult />
        </Provider>
      </BrowserRouter>
    );

    await spyChangePagination(Event, 2)

    //check dispatch one setSearchText after input change
    await wait(() => {
      //test call handleChange
      expect(spyChangePagination).toHaveBeenCalledTimes(1);
      expect(spyChangePagination).toHaveBeenCalledWith(Event, 2);
      //test call searchUsersPage
      expect(spySearchUserPage).toHaveBeenCalledTimes(1);
      expect(spySearchUserPage).toHaveBeenCalledWith(1)
    });
  })

})