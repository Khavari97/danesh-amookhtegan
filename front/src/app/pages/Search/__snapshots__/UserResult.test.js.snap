// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test of ProjectResult page  snapshot 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "display": "flex",
        "margin": "5px 0 40px 0",
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth"
      data-test="placeholder"
      data-testid="search-input"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        defaultValue=""
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="مهارت یا اسم پروژه یا نام کاربری فرد مورد نظر خود را وارد کنید"
        type="text"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-test="btnSearch"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginRight": 5,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        جستجو
        <img
          alt=""
          src="search.svg"
          style={
            Object {
              "marginRight": 5,
            }
          }
          width="100%"
        />
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
        data-test="sorting"
        style={
          Object {
            "padding": "0 5px 15px 0",
            "width": "100%",
          }
        }
      >
         دسته بندی بر اساس : 
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-testid="skills"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 3px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               مهارت 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     مهارت 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-testid="university"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 3px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               دانشگاه 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     دانشگاه 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-testid="province"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 3px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               استان 
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     استان 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-testid="city"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          style={
            Object {
              "marginBottom": "20px",
              "padding": "5px 3px",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="combo-box-demo"
              id="combo-box-demo-label"
            >
               شهر
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="combo-box-demo"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                     شهر
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-root animate__animated animate__fadeIn MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <svg
        height="40"
        viewBox="0 0 65 64.608"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Group 2"
          id="Group_2"
          transform="translate(-1371 -235)"
        >
          <path
            d="M25.006.156c-7.251,0-13.4,4.978-13.4,14.5A24.848,24.848,0,0,0,18.186,31.12c1.47,3.9-1.176,5.36-1.725,5.566C8.878,39.469,0,44.525,0,49.542v1.881c0,6.83,13.013,8.388,25.085,8.388a89.321,89.321,0,0,0,13.248-.862A16.321,16.321,0,0,1,31.513,33a9.781,9.781,0,0,1,.47-1.96H31.9A24.871,24.871,0,0,0,38.49,14.658C38.49,5.134,32.248.156,25.006.156ZM43.9,29.866a13.8,13.8,0,1,0,8.623,24.614l.549.549a2.4,2.4,0,0,0,.392,2.822l6.193,6.193a2.52,2.52,0,0,0,3.527,0l1.1-1.1a2.52,2.52,0,0,0,0-3.527l-6.193-6.193a2.438,2.438,0,0,0-2.9-.392l-.549-.549A13.769,13.769,0,0,0,43.9,29.866Zm0,3.763A10.034,10.034,0,1,1,33.864,43.663,10.028,10.028,0,0,1,43.9,33.629Z"
            fill="#046b00"
            id="icons8_find_user_male"
            transform="translate(1371 234.844)"
          />
          <g
            data-name="Group 1"
            id="Group_1"
            transform="translate(95 -73)"
          >
            <line
              data-name="Line 1"
              fill="none"
              id="Line_1"
              stroke="#046b00"
              strokeLinecap="round"
              strokeWidth="2"
              transform="translate(1314.5 347.5)"
              x2="10.25"
              y2="9.225"
            />
            <line
              data-name="Line 2"
              fill="none"
              id="Line_2"
              stroke="#046b00"
              strokeLinecap="round"
              strokeWidth="2"
              transform="translate(1314.5 347.5)"
              x1="10.25"
              y2="9.225"
            />
          </g>
        </g>
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      >
        کاربری با این مشخصات یافت نشد!
      </p>
    </div>
  </div>,
  <div
    className="MuiGrid-root"
    style={
      Object {
        "display": "table",
        "margin": "0 auto",
        "padding": "40px 0",
      }
    }
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-sizeLarge Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-sizeLarge Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
